migration-cstor-mayastor/replicateddb-restore.md

```diff
@@ -0,0 +1,506 @@
+
+```
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  finalizers:
+  - kubernetes.io/pvc-protection
+  labels:
+    role: mongo
+    velero.io/backup-name: mongo-backup-13-09-23
+    velero.io/restore-name: mongo-restore-13-09-23
+  name: mongodb-persistent-storage-claim-mongod-0
+  namespace: default
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 3Gi
+  storageClassName: mayastor-single-replica
+  volumeMode: Filesystem
+```
+
+## Step 6: Resolve issue where PVC is in a Pending
+
+- Begin by deleting the problematic PVC with the following command:
+
+```
+kubectl delete pvc mongodb-persistent-storage-claim-mongod-0
+```
+
+- Once the PVC has been successfully deleted, you can recreate it using the updated configuration from the `pvc-mongo.yaml` file. Apply the new configuration with the following command:
+
+```
+kubectl apply -f pvc-mongo.yaml
+```
+
+## Step 7: Check Velero init container
+
+After recreating the PVC with Mayastor storageClass, you will observe the presence of a Velero initialization container within the application pod. This container is responsible for restoring the required volumes.
+
+You can check the status of the restore operation by running the following command:
+
+```
+kubectl describe pod <enter_your_pod_name>
+```
+
+![](https://hackmd.io/_uploads/rk1fbgJep.png)
+
+
+The output will display the pods' status, including the Velero initialization container. Initially, the status might show as "Init:0/1," indicating that the restore process is in progress.
+
+You can track the progress of the restore by running:
+
+{% tabs %}
+{% tab title="Command" %}
+```text
+velero get restore
+```
+{% endtab %}
+
+{% tab title="Output" %}
+```text
+NAME                     BACKUP                  STATUS      STARTED                         COMPLETED                       ERRORS   WARNINGS   CREATED                         SELECTOR
+mongo-restore-13-09-23   mongo-backup-13-09-23   Completed   2023-09-13 13:56:19 +0000 UTC   2023-09-13 14:06:09 +0000 UTC   0        4          2023-09-13 13:56:19 +0000 UTC   <none>
+```
+{% endtab %}
+{% endtabs %}
+
+You can then verify the data restoration by accessing your MongoDB instance. In the provided example, we used the "mongosh" shell to connect to the MongoDB instance and check the databases and their content. The data should reflect what was previously backed up from the cStor storage.
+
+```
+mongosh mongodb://admin:admin@mongod-0.mongodb-service.default.svc.cluster.local:27017
+```
+
+
+
+
+## Step 8: Monitor Pod Progress 
+
+Due to the statefulset's configuration with three replicas, you will notice that the `mongo-1` pod is created but remains in a `Pending` status. This behavior is expected as Kubernetes sequentially provisions statefulset pods, ensuring each one is fully operational before moving to the next.
```

Comment:
Due to the statefulset's configuration with three replicas, you will notice that the `mongo-1` pod is created but remains in a `Pending` status. This behavior is expected as we have the storage class set to cStor in statefulset configuration.